app/src/Components/Base/Content.js

```diff
@@ -8,14 +8,51 @@ import DataSource from './../DataSource/DataSource';
 import NotFoundComponent from '../Error/NotFoundComponent'
 import Login from '../Login/Login';
 
+const PrivateRoute = ({ component: Component, ...rest }) => (
+    <Route {...rest} render={(props) => (
+        isAuthentificated(rest)
+            ? <Component {...props} />
+            : <Redirect to={{
+            pathname: '/login',
+            state: { from: props.location }
+        }} />
+    )} />
+);
+
+function isAuthentificated(props){
+    let cookieValue = getCookieValue('token');
+    if(cookieValue !== null && cookieValue.length > 5){
+        let token = parseJwt(cookieValue);
+
+        for(let a = 0; a < props.permissions.length; a++) {
+            if(!token.role.includes(props.permissions[a])) {
```

Comment:
Is `role` an array? If so, call it `roles`?
There is also currently nothing like that in the token?